JavaFXUserInterface/src/main/java/utilities/FormInputValidation.java

```diff
@@ -0,0 +1,300 @@
+            createInputError(textInputControlAnchorPane, FormInputValidation.INPUT_VALIDATION_MESSAGES, manufacturerInput, MAX_20_CHARACTERS_ERROR, index);
+        } else if (!manufacturerInput.getText().trim().matches(regexWordAndHyphen)) {
+            assetValidForm = false;
+            createInputError(textInputControlAnchorPane, FormInputValidation.INPUT_VALIDATION_MESSAGES, manufacturerInput, WORD_HYPHEN_ERROR, index);
+        }
+    }
+
+    private static void serialNumberValidation(AnchorPane textInputControlAnchorPane, TextInputControl serialNumberInput, String regexWordAndHyphen, int index) {
+        if (serialNumberInput.getText().trim().isEmpty()) {
+            assetValidForm = false;
+            createInputError(textInputControlAnchorPane, FormInputValidation.INPUT_VALIDATION_MESSAGES, serialNumberInput, EMPTY_FIELD_ERROR, index);
+        } else if (serialNumberInput.getText().length() > 20) {
+            assetValidForm = false;
+            createInputError(textInputControlAnchorPane, FormInputValidation.INPUT_VALIDATION_MESSAGES, serialNumberInput, MAX_20_CHARACTERS_ERROR, index);
+        } else if (!serialNumberInput.getText().trim().matches(regexWordAndHyphen)) {
+            assetValidForm = false;
+            createInputError(textInputControlAnchorPane, FormInputValidation.INPUT_VALIDATION_MESSAGES, serialNumberInput, WORD_HYPHEN_ERROR, index);
+        }
+    }
+
+    private static void assetDescriptionValidation(AnchorPane textInputControlAnchorPane, TextInputControl assetDescriptionInput, int index) {
+        if (assetDescriptionInput.getText().length() > 300) {
+            assetValidForm = false;
+            createInputError(textInputControlAnchorPane, FormInputValidation.INPUT_VALIDATION_MESSAGES, assetDescriptionInput, MAX_300_CHARACTERS_ERROR, index);
+        }
+    }
+
+    private static void assetNameValidation(AnchorPane textInputControlAnchorPane, TextInputControl assetNameInput, int index) {
+        if (assetNameInput.getText().trim().isEmpty()) {
+            assetValidForm = false;
+            createInputError(textInputControlAnchorPane, FormInputValidation.INPUT_VALIDATION_MESSAGES, assetNameInput, EMPTY_FIELD_ERROR, index);
+        } else if (assetNameInput.getText().length() > 50) {
+            assetValidForm = false;
+            createInputError(textInputControlAnchorPane, FormInputValidation.INPUT_VALIDATION_MESSAGES, assetNameInput, MAX_50_CHARACTERS_ERROR, index);
+        }
+    }
+
+    /**
+     * Compares two thresholds and determines if the previous threshold is larger than the next. If the the previous threshold is larger than
+     * the next, the threshold difference is considered invalid and returns false.
+     *
+     * @param previousThreshold The Threshold preceding
+     * @param nextThreshold     The Threshold succeeding
+     * @author Najim
+     */
+    private static boolean compareThresholds(TextField previousThreshold, TextField nextThreshold) {
```

Comment:
Small warning from Intellij suggesting to invert the method (considered refactoring):
![Screen Shot 2021-03-27 at 2 54 21 PM](https://user-images.githubusercontent.com/46761237/112731250-77f73000-8f0c-11eb-9143-b040232f3837.png)

Reply:
I also saw this and thought about it, but the refactoring of that method was really out of scope, and tbh, at that point, I felt like I did enough refactoring for the in put validation. Because , for this specific method, I simply moved it from where it was to this new class, but I didnt touch it. For me, it works fine. I felt like the effort was enough for this pull request

Reply:
Fair enough, it is a small refactorization that doesn't affect much.